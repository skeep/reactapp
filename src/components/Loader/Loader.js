// @flow

import React from 'react'

type Props = {
  error: Error,
  timedOut: boolean,
  pastDelay: boolean
}

const Loader = (props: Props): any => {
  if (props.error) {
    return <div>Error!</div>
  } else if (props.timedOut) {
    return <div>Taking a long time...</div>
  } else if (props.pastDelay) {
    return <div>Loading...</div>
  } else {
    return null
  }
}

export default Loader
