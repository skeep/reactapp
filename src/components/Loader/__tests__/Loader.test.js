import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Loader from '../Loader.js'

// Snapshot test
it('Snapshot test with content', () => {
  const tree = shallow(<Loader />)
  expect(toJson(tree)).toMatchSnapshot()
})

it('Snapshot test pastDelay', () => {
  const tree = shallow(<Loader pastDelay />)
  expect(toJson(tree)).toMatchSnapshot()
})

it('Snapshot test timedOut', () => {
  const tree = shallow(<Loader timedOut />)
  expect(toJson(tree)).toMatchSnapshot()
})

// Rendering test

// Behaviour tests

// Integration tests

// System tests
