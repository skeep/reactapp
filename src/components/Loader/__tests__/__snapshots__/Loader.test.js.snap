// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test pastDelay 1`] = `
<div>
  Loading...
</div>
`;

exports[`Snapshot test timedOut 1`] = `
<div>
  Taking a long time...
</div>
`;

exports[`Snapshot test with content 1`] = `""`;
